sdk/helper/storagepacker/types.proto

```diff
@@ -0,0 +1,40 @@
+syntax = "proto3";
+
+option go_package = "github.com/hashicorp/vault/sdk/helper/storagepacker";
+
+package storagepacker;
+
+import "google/protobuf/any.proto";
+
+// Item represents an entry that gets inserted into the storage packer
+message Item {
+	// ID must be provided by the caller; the same value, if used with GetItem,
+	// can be used to fetch the item. However, when iterating through a bucket,
+	// this ID will be an internal ID. In other words, outside of the use-case
+	// described above, the caller *must not* rely on this value to be
+	// consistent with what they passed in.
+	string id = 1;
+	// Message holds the contents of the item
+	// Deprecated: Use 'Data' instead
+	google.protobuf.Any message = 2;
```

Comment:
Would it make sense to add a "deprecated" note in the protobuf statement too like is [here](https://developers.google.com/protocol-buffers/docs/proto)? Or is this here for backwards-compatibility?

Reply:
This is copied over from @jefferai's original code so I'm not entirely sure.

I think at one point there may have been some idea of automatically upgrading StoragePacker V1 in this code, which is why it uses the same protobuf definition?

Reply:
I think that would probably not be the right thing to do here.  There doesn't seem to be any support for Go emitting an error, and even if it did that would cause a compiler warning (maybe?) when we do our own check in PutItem.